python/cudf/cudf/core/column/decimal.py

```diff
@@ -96,6 +96,47 @@ def __cuda_array_interface__(self):
             "Decimals are not yet supported via `__cuda_array_interface__`"
         )
 
+    @classmethod
+    def _from_32_64_arrow(
+        cls,
+        data: pa.Array,
+        *,
+        view_type: Literal["int32", "int64"],
+        plc_type: plc.TypeId,
+        step: int,
+    ) -> Self:
+        # Can remove when pyarrow 19 is the minimum version
+        mask_buf, data_buf = data.buffers()
+        rmm_data_buffer = rmm.DeviceBuffer.to_device(
+            np.frombuffer(data_buf)
+            .view(view_type)[::step]
+            .copy()
+            .view("uint8")
+        )
+        plc_column = plc.Column.from_rmm_buffer(
+            rmm_data_buffer,
+            plc.DataType(plc_type, -data.type.scale),
+            len(data),
+            [],
+        )
+        if mask_buf is not None:
+            mask_size = plc.null_mask.bitmask_allocation_size_bytes(len(data))
+            if mask_buf.size < mask_size:
+                rmm_mask_buffer = rmm.DeviceBuffer(size=mask_size)
+                rmm_mask_buffer.copy_from_host(
+                    np.asarray(mask_buf).view("uint8")
+                )
+            else:
+                rmm_mask_buffer = rmm.DeviceBuffer.to_device(
+                    np.frombuffer(mask_buf).view("uint8")
+                )
+            plc_column = plc_column.with_mask(
+                plc.gpumemoryview(rmm_mask_buffer), data.null_count
+            )
+        column = cls.from_pylibcudf(plc_column)
+        column.dtype.precision = data.type.precision
```

Comment:
Just for my understanding, why do we need to set this after the `column` has been created?

Reply:
Since libcudf decimal types don't have a concept of precision, creating a column here via rmm + pylibcudf will [not reflect the same precision](https://github.com/rapidsai/cudf/blob/94e0f922471ebdf4abdd42d7c23644ab33201391/python/cudf/cudf/utils/dtypes.py#L558-L568) as the pyarrow array. In cuDF classic, decimal precision is just kept as metadata on the dtype object.